internal/notifier/adaptive_card.go

```diff
@@ -0,0 +1,194 @@
+/*
+Copyright 2024 The Flux authors
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package notifier
+
+import (
+	"context"
+	"crypto/x509"
+	"fmt"
+	"net/url"
+	"slices"
+	"strings"
+
+	eventv1 "github.com/fluxcd/pkg/apis/event/v1beta1"
+)
+
+// msAdaptiveCardVersion is the version of the MS Adaptive Card schema.
+// MS Teams currently supports only up to version 1.4:
```

Comment:
I think that's a good idea, we can test the address host for the suffix `.webhook.office.com` and send the current payload, otherwise send the Adaptive Card payload. In case we face a use case in the future that also benefits from Adaptive Card, then we can refactor the code and possibly introduce a new provider specific to Adaptive Card.

Reply:
Fixed, thanks!